src/torrent.cpp

```diff
@@ -1138,6 +1138,50 @@ bool is_downloading_state(int const st)
 		if (m_peer_list) m_peer_list->clear();
 	}
 
+	void torrent::set_sequential_range(piece_index_t first_piece, piece_index_t last_piece)
+	{
+		if (!has_picker()) {
+			if (!valid_metadata() || !m_connections_initialized) return;
+			else if (!m_have_all
+				|| settings().get_int(settings_pack::suggest_mode)
+				== settings_pack::suggest_read_cache)
+				need_picker();
+			else return;
+		}
```

Comment:
I think there needs to be a range check for `first_piece` and `last_piece` here, to ensure they form a valid range within the valid piece indices of the torrent. Since this is a call from the client, it should just return as a no-op.

It might even make sense to add a `TORRENT_ASSERT_PRECOND()` too.

Reply:
I don't see why you resolved this comment. I don't see where you validate `first_piece` < num pieces and `last_piece` < num pieces.